docs/docs/business-logic.mdx

```diff
@@ -0,0 +1,322 @@
+---
+id: business-logic
+sidebar_label: Handling Business Logic
+title: Handling Business Logic
+---
+
+<a aria-hidden="true" tabIndex="-1" className="anchor enhancedAnchor" id="handling-business-logic"></a>
+
+A lot of conversational assistants have user goals that involve collecting a bunch of information
+from the user before being able to do something for them. This is called slot filling. For
+example, in the banking industry you may have a user goal of transferring money, where you
+need to collect information about which account to transfer from, whom to transfer to and the
+amount to transfer. This type of behavior can and should be handled in a rule based way, as it
+is clear how this information should be collected.
+
+For this type of use case, we can use Forms and our FormPolicy. The [FormPolicy](./policies.mdx#form-policy)
+works by predicting the form as the next action until all information is gathered from the user.
+
+As an example, we will build out the SalesForm from Sara. The user wants to contact
+our sales team, and for this we need to gather the following pieces of information:
+
+* Their job
+
+* Their bot use case
+
+* Their name
+
+* Their email
+
+* Their budget
+
+* Their company
+
+We will start by defining the `SalesForm` as a new class in the file called `actions.py`.
+The first method we need to define is the name, which like in a regular Action
+returns the name that will be used in our stories:
+
+```python
+from rasa_sdk.forms import FormAction
+
+class SalesForm(FormAction):
+    """Collects sales information and adds it to the spreadsheet"""
+
+    def name(self):
+        return "sales_form"
+```
+
+Next we have to define the `required_slots` method which specifies which pieces of information to
+ask for, i.e. which slots to fill.
+
+```python
+@staticmethod
+def required_slots(tracker):
+    return [
+        "job_function",
+        "use_case",
+        "budget",
+        "person_name",
+        "company",
+        "business_email",
+        ]
+```
+
+Note: you can customise the required slots function not to be static. E.g. if the `job_function` is a
```

Comment:
typoCI assumes we want the US english version. Is it right to assume this?

Reply:
Yes, we decided when we included typoci to do US-English only